pkg/steps/release/promote.go

```diff
@@ -94,11 +92,11 @@ func (s *promotionStep) run(ctx context.Context) error {
 	if len(s.config.Name) > 0 {
 		return retry.RetryOnConflict(promotionRetry, func() error {
 			is := &imagev1.ImageStream{}
-			err := s.dstClient.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: s.config.Namespace, Name: s.config.Name}, is)
+			err := s.client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: s.config.Namespace, Name: s.config.Name}, is)
 			if errors.IsNotFound(err) {
 				is.Namespace = s.config.Namespace
 				is.Name = s.config.Name
-				if err := s.dstClient.Create(ctx, is); err != nil {
+				if err := s.client.Create(ctx, is); err != nil {
```

Comment:
This is a podclient right? This line will create an imagestream.

Reply:
@droslean a pod client looks like this: https://github.com/openshift/ci-tools/blob/e51f853f419026ba115871cfac4c224f7d4b1180/pkg/steps/artifacts.go#L178-L182

It can do CRUD for anything and additionally has methods for the `Exec/GetLogs` pod subresources